rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+  outgoing_result_response.sequence.data[3] = 6;
+  outgoing_result_response.status =
+    action_msgs__msg__GoalStatus__STATUS_SUCCEEDED;
+
+  // Initialize cancel request
+  rmw_request_id_t cancel_request_header;
+  init_test_uuid0(outgoing_cancel_request.goal_info.goal_id.uuid);
+  outgoing_cancel_request.goal_info.stamp.sec = 321;
+  outgoing_cancel_request.goal_info.stamp.nanosec = 987654u;
+
+  // Send cancel request with valid arguments
+  ret = rcl_action_send_cancel_request(
+    &this->action_client, &outgoing_cancel_request, &sequence_number);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait_set_clear(&this->wait_set);
+
+  ret = rcl_action_wait_set_add_action_server(&this->wait_set, &this->action_server, NULL);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait(&this->wait_set, 1000000);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_server_wait_set_get_entities_ready(
+    &this->wait_set,
+    &this->action_server,
+    &this->is_goal_request_ready,
+    &this->is_cancel_request_ready,
+    &this->is_result_request_ready,
+    &this->is_goal_expired);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  EXPECT_TRUE(this->is_cancel_request_ready);
+  EXPECT_FALSE(this->is_goal_request_ready);
+  EXPECT_FALSE(this->is_result_request_ready);
+
+  // Take cancel request with valid arguments
+  ret = rcl_action_take_cancel_request(
+    &this->action_server, &cancel_request_header, &incoming_cancel_request);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Check that the cancel request was received correctly
+  EXPECT_TRUE(uuidcmp(
+      outgoing_cancel_request.goal_info.goal_id.uuid,
+      incoming_cancel_request.goal_info.goal_id.uuid));
+  EXPECT_EQ(
+    outgoing_cancel_request.goal_info.stamp.sec,
+    incoming_cancel_request.goal_info.stamp.sec);
+  EXPECT_EQ(
+    outgoing_cancel_request.goal_info.stamp.nanosec,
+    incoming_cancel_request.goal_info.stamp.nanosec);
+
+  rcl_action_cancel_request_t cancel_request = rcl_action_get_zero_initialized_cancel_request();
+  for (size_t i = 0; i < 16; ++i) {
+    cancel_request.goal_info.goal_id.uuid[i] = incoming_cancel_request.goal_info.goal_id.uuid[i];
+  }
+  cancel_request.goal_info.stamp.sec = incoming_cancel_request.goal_info.stamp.sec;
+  cancel_request.goal_info.stamp.nanosec = incoming_cancel_request.goal_info.stamp.nanosec;
+
+
+  // Get a list of goal info that should be attempted to be cancelled
+  rcl_action_cancel_response_t cancel_response = rcl_action_get_zero_initialized_cancel_response();
+  ret = rcl_action_process_cancel_request(
+    &this->action_server, &cancel_request, &cancel_response);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Initialize cancel response
```

Comment:
Do we really need to do this, or can we use `cancel_response.msg` directly? Perhaps asserting some things about it first, e.g. `EXPECT_EQ(cancel_response.msg.goals_canceling.size, 2u)`.

Reply:
Just removed that initialization part